addons/im_livechat/static/src/core/web/thread_service_patch.js

```diff
@@ -59,4 +59,10 @@ patch(ThreadService.prototype, {
         this.chatWindowService.focus(chatWindow);
         return true;
     },
+
+    pin(thread) {
+        if (!this.store.discuss.fromLivechatMenu) {
+            super.pin(thread);
+        }
+    },
```

Comment:
unreliable override, I would not do that, and make the base code of pin be called/do the appropriate code only when it makes sense